### PR TITLE
fix(rfi): additional protocols for RFI rule 931130

### PR DESCRIPTION
**Pull request by karelorigin**
This PR contains fixes for bypasses that were originally found and submitted by @hussein98d in `9LB1ZNRE`, `V4DR54LM` and `J27G5EJX`.

### CODE REVIEW

**Comment by lifeforms** [comment]
> I noticed that many SSRF/RFI rules are only evaluated against query parameters. Perhaps including XML, headers, and other common injection points is also a good idea?

I agree.

Our generic RFI rule (931130) can't run on request headers, it will fire on most http:// links and this will give false positives on the Referer and the User-Agent request headers.

However, for specific SSRF rules, the patterns are pretty much a sure indicative of an attack, and false positives don't seem so much of a problem, so I agree we should check all the variables.

I think the best way is to make a new rule with the other protocols (except http/https) and have that rule cover all variables including headers and XML.

**Comment by karelorigin** [comment]
I've just pushed another fix for a bypass submitted by @hussein98d in `YJGXRICC`

**Comment by dune73** [comment]
### CRS Bug Bounty PR assessment

* **Rules affected** (list rules): FIXME
* **Paranoia Level addressed** (1, 2, 3, 4, full or explain): FIXME
* **FTW passes** (yes or no) : FIXME
* **Rule(s) picked for solution** (correct or not-correct or explain) : FIXME
* **Risk for false positives** (irrelevant, adequate, substantial or explain) : FIXME
* **Regular expression quality** (inspirational, decent base, needs work, adequate or explain) : FIXME
* **Documentation** (needs work, adequate or explain) : FIXME
* **Tests** (none or some or adequate) : FIXME
&nbsp;
* **Verdict** (Unusable, inspirational, usable, almost perfect or perfect) : FIXME

_This is not meant to be final. As a CRS dev, feel free to comment below and edit this form directly. As committer or observer, feel free to comment below with feedback and we will think about updating the assessment accordingly._

**Comment by lifeforms** [comment]
`ldaps` is also a protocol (secure LDAP), `ldapi` is also there. Could we try to come up with an exhaustive list of protocols? I think there were also a bunch of "PHP wrappers" posted in a submission, which are basically similar to protocols.

**Comment by fzipi** [comment]
Now that we merged the 934120 update, you can use the same technique used there in lines https://github.com/coreruleset/coreruleset/blob/51b5d465d9b7db3ec7654762165eafaa6c9b79ae/util/regexp-assemble/data/934120.data#L14-L19  to include the url-schemes list.

**Comment by dune73** [comment]
Assigning @fzipi based on our separate spreadsheet.

**Comment by fzipi** [comment]
@karelorigin Just a question: are you going to have time to update this one, or do you want me to add the latest changes? I know you are very busy these days...

**Comment by karelorigin** [comment]
Sorry for the wait! I should be able to get the necessary work done before the weekend is over. Is the include directive global? Can I reference it in other data files? @fzipi

**Comment by karelorigin** [comment]
Awesome @fzipi, I'll get on that as soon as I can.

**Comment by fzipi** [comment]
@karelorigin Hey! Any news on this one? Do you need help?

**Comment by fzipi** [comment]
Solved conflicts.

**Comment by fzipi** [comment]
Fixed also:

- moved to including url-schemes
- fixed `file://` inclusion as url-scheme 🤦
- fixed dependent rule 934120.

File: rules/REQUEST-934-APPLICATION-ATTACK-GENERIC.conf [hunk 1]
```diff
@@ -238,7 +238,7 @@ SecRule TX:DETECTION_PARANOIA_LEVEL "@lt 2" "id:934014,phase:2,pass,nolog,skipAf
 # (consult util/regexp-assemble/README.md for details):
 #   util/regexp-assemble/regexp-assemble.py update
```
Comment (theseion) [inline rules/REQUEST-934-APPLICATION-ATTACK-GENERIC.conf:239]:
```suggestion
#   util/regexp-assemble/regexp-assemble.py update 934120
```

File: rules/REQUEST-931-APPLICATION-ATTACK-RFI.conf [hunk 2]
```diff
@@ -107,7 +107,7 @@ SecRule TX:DETECTION_PARANOIA_LEVEL "@lt 2" "id:931014,phase:2,pass,nolog,skipAf
 
 # url:file:// can be used by Java applications using
 # org.apache.commons.io.IOUtils to access internal files, so this has been added
```
Comment (theseion) [inline rules/REQUEST-931-APPLICATION-ATTACK-RFI.conf:109]:
```suggestion
# org.apache.commons.io.IOUtils to access internal files, so this has been added
#
#   # Regular expression generated from util/regexp-assemble/data/931130.data.
# To update the regular expression run the following shell script
# (consult util/regexp-assemble/README.md for details):
#   util/regexp-assemble/regexp-assemble.py update 931130
#
```

**Review by theseion — CHANGES_REQUESTED** [review]

**Review by theseion — APPROVED** [review]
LGTM!

**Comment by fzipi** [comment]
Thanks @karelorigin for pushing this change!

**Pull request merged**